iotilecore/RELEASE.md

```diff
@@ -2,6 +2,11 @@
 
 All major changes in each released version of `iotile-core` are listed here.
 
+## 5.0.3
+
+- JLink adapter support
```

Comment:
I don't think there's any jlink specific change in here so that release note isn't needed unless I'm missing it.

Reply:
Keeping 5.0.3 version bump, removing JLINK note. Fixed.